tests/unit_tests/test_apply_dimension.py

```diff
@@ -0,0 +1,186 @@
+                    {"B01": 1, "B02": 2},
+                    {"B01": 3, "B02": 4},
+                    {"B01": 5, "B02": 6},
+                ],
+                "process": "({data}) => data.map(el => el * 3)",
+                "dimension": "bands",
+            },
+            [3, 6, 9, 12, 15, 18],
+        ),
+        (
+            {
+                "data": [
+                    {"B01": 1, "B02": 2},
+                    {"B01": 3, "B02": 4},
+                    {"B01": 5, "B02": 6},
+                ],
+                "process": "({data}) => {const sum = data.reduce((a, b) => a + b, 0); return data.map(el => el * sum)}",
+                "dimension": "bands",
+            },
+            [3, 6, 21, 28, 55, 66]
+        ),
+    ],
+)
+def test_apply_dimension(apply_dimension_process_code, example_input, expected_result):
+    additional_js_code_to_run = (
+        load_datacube_code() + f"const cube = new DataCube({example_input['data']}, 'bands', 'temporal', true);"
+    )
+    process_arguments = f"{{...{json.dumps(example_input)}, 'data': cube, 'process':{example_input['process']}}}"
+    output = run_process(
+        apply_dimension_process_code + additional_js_code_to_run,
+        "apply_dimension",
+        process_arguments,
+    )
+    output = json.loads(output)
+
+    assert output["data"]["data"] == expected_result
+
+
+@pytest.mark.parametrize(
+    "example_input,raises_exception,error_name",
+    [
+        (
+            {
+                "data": [{ "B01": 1, "B02": 2, "B03": 3}],
+                "process": "({data})=>1",
+                "dimension": "bands",
+            },
+            False,
+            None,
+        ),
+        (
+            {
+                "process": "({data})=>1",
+                "dimension": "bands",
+            },
+            True,
+            "MISSING_PARAMETER",
+        ),
+        (
+            {
+                "data": [{ "B01": 1, "B02": 2, "B03": 3}],
+                "dimension": "bands",
+            },
+            True,
+            "MISSING_PARAMETER",
+        ),
+        (
+            {
+                "data": [{ "B01": 1, "B02": 2, "B03": 3}],
+                "process": "({data})=>1",
+            },
+            True,
+            "MISSING_PARAMETER",
+        ),
+        (
+            {
+                "data": [{ "B01": 1, "B02": 2, "B03": 3}],
+                "process": "({data})=>1",
+                "dimension": True,
+            },
+            True,
+            "WRONG_TYPE",
+        ),
+        (
+            {
+                "data": [{ "B01": 1, "B02": 2, "B03": 3}],
+                "process": "({data})=>1",
+                "dimension": "bands",
+                "target_dimension": True,
+            },
+            True,
+            "WRONG_TYPE",
+        ),
+    ],
+)
+def test_input_validation(apply_dimension_process_code, example_input, raises_exception, error_name):
+
+    data = example_input["data"] if "data" in example_input else None
```

Comment:
Minor: `example_input.get("data")` does the same thing. Similarly next line. fftr.